cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -409,22 +441,36 @@ Status TransferDate64(RecordReader* reader, MemoryPool* pool,
 
 Status TransferDictionary(RecordReader* reader,
                           const std::shared_ptr<DataType>& logical_value_type,
-                          std::shared_ptr<ChunkedArray>* out) {
+                          bool nullable, std::shared_ptr<ChunkedArray>* out) {
   auto dict_reader = dynamic_cast<DictionaryRecordReader*>(reader);
   DCHECK(dict_reader);
   *out = dict_reader->GetResult();
   if (!logical_value_type->Equals(*(*out)->type())) {
     ARROW_ASSIGN_OR_RAISE(*out, (*out)->View(logical_value_type));
   }
+  if (!nullable) {
+    // Reconstruct each chunk without nulls.
```

Comment:
Is this necessary for correctness or is it just an optimization of eliding the null buffer?

Reply:
Is there a situation where said null buffer would not be all ones?

Reply:
Hmm, I see, thank you.